### PR TITLE
Add metadata only once for a repository

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
CentralDogma currently adds metadata information of a repository into metadata.json if it doesn't exist.
However, if `MetadataService.getProject(...) is called multiple times before the metadata is added, there can be multiple commits of adding operation.
(We have watched over 1000 times for the same repository.)

Modifications:
- Fix to commit only once for the metadata of a repository.

Result:
- You no longer see multiple `Adding missing repository metadata` for the same repository.

Todo:
- Change to add metadata after `CreateRepository` is called.

### CODE REVIEW

**Review by trustin — APPROVED** [review]
Thanks, @minwoox and @ikhoon :smile:

File: server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java [hunk 1]
```diff
@@ -119,8 +123,26 @@ public MetadataService(ProjectManager projectManager, CommandExecutor executor)
                     continue;
                 }
 
+                final CompletableFuture<Revision> future = new CompletableFuture<>();
+                final CompletableFuture<Revision> futureInMap =
+                        reposInAddingMetadata.computeIfAbsent(repo, key -> future);
```
Comment (ikhoon) [inline server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java:128]: Don't we need to include `projectName` in the cache key?

Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java:128]: Oops thanks!

Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java]: Fixed. 😉

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox 🙇‍♂️

**Comment by minwoox** [comment]
Thanks @trustin and @ikhoon!

**Pull request merged**